cmd/sourced/cmd/compose.go

```diff
@@ -66,18 +68,28 @@ type composeSetDefaultCmd struct {
 	Command `name:"set" short-description:"Set the active docker compose file" long-description:"Set the active docker compose file"`
 
 	Args struct {
-		Version string `positional-arg-name:"version" description:"Either a revision (tag, full sha1) or a URL to a docker-compose.yml file"`
+		Index string `positional-arg-name:"index" description:"Index of the docker compose file returned from 'sourced compose list'"`
```

Comment:
maybe it makes sense to accept both the index and the name?
I see why index can be better in many cases but at the same time, it's easier to switch to `master` just by typing `sourced compose set master` instead of listing the files and looking for the correct number.
what do you think?

Reply:
Do you mean only for master or all rev and URL?

Reply:
I think that it should work as now + with index. So essentially the tests that are already in `master` should pass without any change, then it should also work with indexes.

Reply:
I guess that it's safe to try parsing whether it's an integer first and in case of a parsing error fallback to the current mechanism.

Reply:
What if there is a branch, or tag called `1`?

Reply:
That's definitely a case that I thought, but it's really and edge case that IMHO could be ignored. I don't think that it makes much sense to have a branch named `\d+`. But if you prefer you may want to default as integer and provide a flag `--rev` to force it. But again IMHO it's unnecessary.

Reply:
We control tags and branches so there is no need to add additional logic for the edge case.

Reply:
What git does, for example when performing a checkout to a revision that can be resolved to different commits, is to rise an error, and ask for a more precise definition of the revision.

Reply:
Changed it to both accept index and name by ignoring the edge cases.